uniffi_meta/src/lib.rs

```diff
@@ -308,6 +308,7 @@ pub struct FieldMetadata {
 pub struct EnumMetadata {
     pub module_path: String,
     pub name: String,
+    pub forced_flatness: Option<bool>,
```

Comment:
Not that important, but I wonder if we could avoid adding this field by always writing out the non-flat version of the metadata and ensuring that each variant has 0 fields.

Reply:
Unfortunately I don't think we can, for the same reason this must be `Option<bool>` and not a plain bool. In some cases where this is `Some(true)` the variant *is not* flat and we still need the variants written. `Some(true)` is used just by foreign bindings and says "even though this isn't flat, use the same capitalization and other conventions etc as though it was"

Let me know if this doesn't make sense or if I'm confused!

Reply:
That makes sense to me.  I don't think your confused, but I still wonder if there's a better way to handle it.  That doesn't really concern this PR though and shouldn't block it.